--- "povray-3.8.0-beta.2\\source\\parser\\dump.h"	1970-01-01 04:00:00.000000000 +0400
+++ "povdump-3.8.0-beta.2\\source\\parser\\dump.h"	2022-12-17 23:30:11.260995400 +0400
@@ -0,0 +1,9 @@
+#ifndef POVRAY_PARSER_DUMP_H
+#define POVRAY_PARSER_DUMP_H
+
+#define DUMP(b,a) fwrite(&(a),sizeof(a),1,(b))
+#define DUMPTAG(b,a) fwrite(&(a),4,1,(b))
+#define TMPCFG_NAME "scene.cfg"
+#define SCENE_DUMP_NAME "scene.dump"
+
+#endif // POVRAY_PARSER_DUMP_H
