--- "povray-3.8.0-beta.2\\source\\parser\\parser.cpp"	2021-08-09 05:11:00.000000000 +0400
+++ "povdump-3.8.0-beta.2\\source\\parser\\parser.cpp"	2022-12-18 22:09:36.893423200 +0400
@@ -37,7 +37,7 @@
 
 // Unit header file must be the first file included within POV-Ray *.cpp files (pulls in config)
 #include "parser/parser.h"
-
+#include "parser/dump.h"
 // C++ variants of standard C header files
 #include <cctype>
 #include <cmath>
@@ -520,6 +520,8 @@ void Parser::Frame_Init()
     sceneData->fog = nullptr;
     sceneData->rainbow = nullptr;
     sceneData->skysphere = nullptr;
+
+	Log("Frame_Init()");
 }
 
 
@@ -611,6 +613,8 @@ void Parser::InitDefaults(POVRayVersion
 
 void Parser::Destroy_Frame()
 {
+	Log("Destroy_Frame()");
+
     FOG *Fog, *Next_Fog;
     RAINBOW *Rainbow, *Next_Rainbow;
 
@@ -10336,6 +10340,7 @@ void Parser::Link_To_Frame(ObjectPtr Obj
         (dynamic_cast<CSGMerge *>(Object) != nullptr))
     {
         Link(Object, sceneData->objects);
+		Log("link 1");
         return;
     }
 
@@ -10352,6 +10357,7 @@ void Parser::Link_To_Frame(ObjectPtr Obj
     if((reinterpret_cast<CSGUnion *>(Object))->do_split == false)
     {
         Link(Object, sceneData->objects);
+		Log("link 2");
         return;
     }
 
@@ -10380,6 +10386,7 @@ void Parser::Link_To_Frame(ObjectPtr Obj
             if (finite)
                 Warning("CSG union unnecessarily bounded.");
             Link(Object, sceneData->objects);
+			Log("link 3");
             return;
         }
 
@@ -10395,6 +10402,7 @@ void Parser::Link_To_Frame(ObjectPtr Obj
     }
 
     (reinterpret_cast<CSG *>(Object))->children.clear();
+	Log("Destroy_Object(Object)");
     Destroy_Object(Object);
 }
 
