--- "povray-3.8.0-beta.2\\source\\base\\build.h"	2022-12-14 17:13:48.540783000 +0400
+++ "povdump-3.8.0-beta.2\\source\\base\\build.h"	2022-12-14 13:40:38.247486800 +0400
@@ -125,8 +125,8 @@
 ///     undesired interference between the repository and your local changes.
 ///
 #ifndef BUILT_BY
-    #define BUILT_BY "YOUR NAME (YOUR EMAIL)"
-    #error "Please fill in BUILT_BY, then remove this line"
+    #define BUILT_BY "slair (slair@github.com)"
+    //#error "Please fill in BUILT_BY, then remove this line"
 #endif
 
 /// @}
